File: server/datasets/python_source-code/Multiple-G1/renamed_47_bst.py
class Node_11:
    def __init__(self, val_11):
        self.val_11 = val_11
        self.lhs_11 = None
        self.rhs_11 = None

class Tree_11:
    def __init__(self):
        self.rt_11 = None

    def ins_11(self, val_11):
        if self.rt_11 is None:
            self.rt_11 = Node_11(val_11)
        else:
            self._ins_11_rec(self.rt_11, val_11)

    def _ins_11_rec(self, nd_11, val_11):
        if val_11 < nd_11.val_11:
            if nd_11.lhs_11 is None:
                nd_11.lhs_11 = Node_11(val_11)
            else:
                self._ins_11_rec(nd_11.lhs_11, val_11)
        elif val_11 > nd_11.val_11:
            if nd_11.rhs_11 is None:
                nd_11.rhs_11 = Node_11(val_11)
            else:
                self._ins_11_rec(nd_11.rhs_11, val_11)

    def srch_11(self, val_11):
        return self._srch_11_rec(self.rt_11, val_11)

    def _srch_11_rec(self, nd_11, val_11):
        if nd_11 is None or nd_11.val_11 == val_11:
            return nd_11
        if val_11 < nd_11.val_11:
            return self._srch_11_rec(nd_11.lhs_11, val_11)
        return self._srch_11_rec(nd_11.rhs_11, val_11)

File: server/datasets/controlled_benchmarks/java/multi_student_cohorts/cohort_g1/renamed_47_bst.java
class Node_11 {
    int val_11;
    Node_11 lhs_11, rhs_11;
    public Node_11(int item) {
        val_11 = item;
        lhs_11 = rhs_11 = null;
    }
}
public class Tree_11 {
    public static void main(String[] args) {
        Tree_11 bst = new Tree_11();
        bst.ins_11(50);
        bst.srch_11(50);
    }
    Node_11 rt_11;
    public void ins_11(int val_11) {
        rt_11 = in_rec_11(rt_11, val_11);
    }
    Node_11 in_rec_11(Node_11 rt_11, int val_11) {
        if (rt_11 == null) return new Node_11(val_11);
        if (val_11 < rt_11.val_11) rt_11.lhs_11 = in_rec_11(rt_11.lhs_11, val_11);
        else if (val_11 > rt_11.val_11) rt_11.rhs_11 = in_rec_11(rt_11.rhs_11, val_11);
        return rt_11;
    }
    Node_11 srch_11(int val_11) {
        return sr_rec_11(rt_11, val_11);
    }
    Node_11 sr_rec_11(Node_11 rt_11, int val_11) {
        if (rt_11 == null || rt_11.val_11 == val_11) return rt_11;
        if (rt_11.val_11 < val_11) return sr_rec_11(rt_11.rhs_11, val_11);
        return sr_rec_11(rt_11.lhs_11, val_11);
    }
}